clang/include/clang/Basic/AttrDocs.td

```diff
@@ -2098,7 +2098,7 @@ This attribute can only be applied to functions and indicates that the
 function must be treated as a device function and must be emitted even if it has
 no direct uses from other SYCL device functions. However, it cannot be applied
 to functions marked as 'static', functions declared within an anonymous
-namespace or class member functions.
+namespace or virtual/pure virtual class member functions.
```

Comment:
I'm not sure the virtual/pure virtual is worth the differentiation.  I'd say simply 'virtual' is sufficient.  Same in the error message.